Classes/Utility/FrontendUtility.php

```diff
@@ -185,21 +185,21 @@ public static function isCacheable($conf = null)
     /**
      * Return current URL
      *
-     * @return null|string
+     * @return string
      */
     public static function getCurrentUrl()
     {
-        $ret = null;
-
-        $TSFE = self::getTsfe();
+        //former $TSFE->anchorPrefix got deprecated in TYPO3 7.x
+        $tsfeAnchorPrefix = substr(
+            Typo3GeneralUtility::getIndpEnv('TYPO3_REQUEST_URL'),
```

Comment:
Hm... delivering the current Url is the kind of task a CMS should provide. Caching included.